Modules/Core/Common/include/itkPoint.h

```diff
@@ -365,11 +365,11 @@ template <typename TValue, typename... TVariadic>
 auto
 MakePoint(const TValue firstValue, const TVariadic... otherValues)
 {
-  constexpr unsigned int dimension{ 1 + sizeof...(TVariadic) };
-
-  // Note that the class template argument deduction guide of std::array ensures that all values have the same type.
-  const std::array stdArray{ firstValue, otherValues... };
+  static_assert(std::conjunction_v<std::is_same<TVariadic, TValue>...>,
```

Comment:
Instead of `std::conjunction_v`, we could also use a C++17 [_fold expression_](https://en.cppreference.com/w/cpp/language/fold), for example:

    static_assert((std::is_same_v<TVariadic, TValue> && ...));